exchanges/orderbook/orderbook.go

```diff
@@ -15,71 +14,71 @@ import (
 
 // Get checks and returns the orderbook given an exchange name and currency pair
 func Get(exchange string, p currency.Pair, a asset.Item) (*Base, error) {
-	return service.Retrieve(exchange, p, a)
+	return s.Retrieve(exchange, p, a)
 }
 
 // GetDepth returns a Depth pointer allowing the caller to stream orderbook changes
 func GetDepth(exchange string, p currency.Pair, a asset.Item) (*Depth, error) {
-	return service.GetDepth(exchange, p, a)
+	return s.GetDepth(exchange, p, a)
 }
 
 // DeployDepth sets a depth struct and returns a depth pointer. This allows for
 // the loading of a new orderbook snapshot and incremental updates via the
 // streaming package.
 func DeployDepth(exchange string, p currency.Pair, a asset.Item) (*Depth, error) {
-	return service.DeployDepth(exchange, p, a)
+	return s.DeployDepth(exchange, p, a)
 }
 
 // SubscribeToExchangeOrderbooks returns a pipe to an exchange feed
 func SubscribeToExchangeOrderbooks(exchange string) (dispatch.Pipe, error) {
-	service.mu.Lock()
-	defer service.mu.Unlock()
-	exch, ok := service.books[strings.ToLower(exchange)]
+	s.m.RLock()
+	defer s.m.RUnlock()
+	id, ok := s.exchangeIDs[exchange]
 	if !ok {
 		return dispatch.Pipe{}, fmt.Errorf("%w for %s exchange", ErrOrderbookNotFound, exchange)
 	}
-	return service.Mux.Subscribe(exch.ID)
+	return s.signalMux.Subscribe(id)
 }
 
 // Update stores orderbook data
-func (s *Service) Update(b *Base) error {
-	name := strings.ToLower(b.Exchange)
-	mapKey := key.PairAsset{
-		Base:  b.Pair.Base.Item,
-		Quote: b.Pair.Quote.Item,
-		Asset: b.Asset,
-	}
-
-	s.mu.Lock()
-	m1, ok := s.books[name]
+func (s *store) Update(b *Base) error {
+	s.m.RLock()
+	book, ok := s.orderbooks[key.ExchangePairAsset{Exchange: b.Exchange, Base: b.Pair.Base.Item, Quote: b.Pair.Quote.Item, Asset: b.Asset}]
+	s.m.RUnlock()
 	if !ok {
-		id, err := s.Mux.GetID()
+		var err error
+		book, err = s.trackNewBook(b)
 		if err != nil {
-			s.mu.Unlock()
 			return err
 		}
-		m1 = Exchange{
-			m:  make(map[key.PairAsset]*Depth),
-			ID: id,
-		}
-		s.books[name] = m1
 	}
-	book, ok := m1.m[mapKey]
-	if !ok {
-		book = NewDepth(m1.ID)
-		book.AssignOptions(b)
-		m1.m[mapKey] = book
-	}
-	err := book.LoadSnapshot(b.Bids, b.Asks, b.LastUpdateID, b.LastUpdated, b.UpdatePushedAt, true)
-	s.mu.Unlock()
-	if err != nil {
+	if err := book.Depth.LoadSnapshot(b.Bids, b.Asks, b.LastUpdateID, b.LastUpdated, b.UpdatePushedAt, true); err != nil {
 		return err
 	}
-	return s.Mux.Publish(book, m1.ID)
+	return s.signalMux.Publish(book.Depth, book.ExchangeID)
+}
+
+func (s *store) trackNewBook(b *Base) (bookWithExchangeID, error) {
```

Comment:
🐒 This is a store of books, so can just be `track`
```suggestion
func (s *store) track(b *Base) (bookWithExchangeID, error) {
```